app/preprints/-components/preprint-provider-selection/component.ts

```diff
@@ -0,0 +1,44 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { serviceLinks } from 'ember-osf-web/const/service-links';
+import { tracked } from '@glimmer/tracking';
+import Component from '@glimmer/component';
+import PreprintProviderModel from 'ember-osf-web/models/preprint-provider';
+import RouterService from '@ember/routing/router-service';
+
+
+interface InputArgs {
+    allProviders: PreprintProviderModel[];
+}
+
+
+export default class PreprintProviderSelection extends Component<InputArgs> {
+    @service router!: RouterService;
+
+    allProviders: PreprintProviderModel[] = this.args.allProviders;
+    learnMoreUrl: string = serviceLinks.preprintsSupport;
+    @tracked selectedProvider: PreprintProviderModel | undefined;
+
+    public get isDisabled(): boolean {
+        return this.selectedProvider === undefined;
+    }
+
+    public get selectedProviderId(): string | undefined {
+        if (this.selectedProvider !== undefined) {
```

Comment:
Similarly, there's nothing wrong with this code, but it can be shortened to just be 
```
        if (this.selectedProvider) {
```
as the `if ()` statement just checks for ["truthiness"](https://developer.mozilla.org/en-US/docs/Glossary/Truthy)